lib/hedwig_sms.ex

```diff
@@ -8,7 +8,7 @@ defmodule Hedwig.Adapters.SMS do
   @doc false
   def init({robot, opts}) do
     HTTPoison.start
-    Hedwig.Robot.register(robot, opts[:name])
+    :global.register_name(opts[:name], robot)
```

Comment:
You probably want to use `"#{opts[:name]}_adapter"`, otherwise its highly likely that you may end up with a name conflict if the user registers their robot in `handle_connect/1` with the name of their bot.